weather_mv/loader_pipeline/pipeline.py

```diff
@@ -57,6 +57,16 @@ def _prod(xs: t.Iterable[int]) -> int:
     return reduce(operator.mul, xs, 1)
 
 
+def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
```

Comment:
I don't think this function is necessary. xarray's `open_dataset` provides the `cache` argument, which can be used to open data in memory: https://xarray.pydata.org/en/stable/generated/xarray.open_dataset.html

Reply:
The `chunks` argument may also help here.

Reply:
cache=True is lazy caching, while it does improve read times, its much smaller improvement (only ~2x) on the data i tested. 
Unless we have a memory concern, don't think using that path instead is much more helpful.

Reply:
I see. I'm glad that you tested this pathway. In that case, I'm happy with the solution you provided. I'll take one more look at the CL to see if I can find any issues.